salt/cloud/clouds/digital_ocean.py

```diff
@@ -694,17 +738,25 @@ def destroy(name, call=None):
     data = show_instance(name, call='action')
     node = query(method='droplets', droplet_id=data['id'], http_method='delete')
 
-    delete_record = config.get_cloud_config_value(
-        'delete_dns_record', get_configured_provider(), __opts__, search_global=False, default=None,
-    )
+    ## This is all terribly optomistic:
+    # vm_ = get_vm_config(name=name)
+    # delete_dns_record = config.get_cloud_config_value(
+    #     'delete_dns_record', vm_, __opts__, search_global=False, default=None,
+    # )
+    delete_dns_record = True
 
-    if delete_record and not isinstance(delete_record, bool):
+    if delete_dns_record and not isinstance(delete_dns_record, bool):
         raise SaltCloudConfigError(
             '\'delete_dns_record\' should be a boolean value.'
         )
 
-    if delete_record:
-        delete_dns_record(name)
+    if delete_dns_record:
```

Comment:
I should add comments. I really struggled to find a way to get configuration data so that delete_record can be asserted by configuration instead of code, but I don't see a good way, and implementing this should be a separate issue/enhancement/bug.